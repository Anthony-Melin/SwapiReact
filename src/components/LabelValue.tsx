import { PropsWithChildren } from 'react'

type LabelValueProps = PropsWithChildren<{
    label: string
    value: string
    testid: string
    remap?: Record<string, string>
}>

const LabelValue = ({
    label,
    value,
    testid,
    remap = {},
    ...restProps
}: LabelValueProps) => {
    const displayedValue = value in remap ? remap[value] : value
    return (
        <p data-testid={testid} {...restProps}>
            {label}: {displayedValue}
        </p>
    )
}

export default LabelValue
