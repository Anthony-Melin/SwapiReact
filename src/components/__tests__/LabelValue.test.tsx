import { describe, it, expect } from 'vitest'
import { render, screen } from '@testing-library/react'
import LabelValue from '../LabelValue.tsx'

describe('LabelValue', () => {
    it('should match snapshot', () => {
        render(<LabelValue label="LABEL" value="VALUE" testid="testid" />)
        expect(screen.getByTestId('testid')).toMatchSnapshot()
    })

    it('should render text', () => {
        render(<LabelValue label="LABEL" value="VALUE" testid="testid" />)
        expect(screen.getByTestId('testid')).toHaveTextContent('LABEL: VALUE')
    })

    it('should render remaped value', () => {
        render(
            <LabelValue
                label="LABEL"
                value="VALUE"
                remap={{ VALUE: 'REMAP_VALUE' }}
                testid="testid"
            />
        )
        expect(screen.getByTestId('testid')).toHaveTextContent(
            'LABEL: REMAP_VALUE'
        )
    })
})
