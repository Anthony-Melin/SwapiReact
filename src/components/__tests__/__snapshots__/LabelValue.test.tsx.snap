// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LabelValue > should match snapshot 1`] = `
<p
  data-testid="testid"
>
  LABEL
  : 
  VALUE
</p>
`;
